tools/update_runnablejobs.py

```diff
@@ -0,0 +1,113 @@
+                json.dump(data, f)
+    else:
+        print "It's going to help you download the runnable jobs file."
+        data = query_the_runnablejobs(new_timestamp, task_id)
+        if data:
+            with open(ROOT_DIR + '/runnablejobs.json', 'w') as f:
+                json.dump(data, f)
+
+
+def query_the_runnablejobs(new_timestamp, task_id=None):
+    if task_id:
+        url = TREEHERDER_HOST.format('mozilla-inbound', task_id)
+        data = retry(requests.get, args=(url, ), kwargs={'headers': headers}).json()
+        if len(data['results']) > 0:
+            data['meta'].update({'timetamp': new_timestamp})
+            return data
+
+
+def add_new_jobs_into_pressed(new_data=None):
+    if new_data:
+        with open(ROOT_DIR + '/runnablejobs.json', 'r+') as f:
+            old_data = json.loads(f.read())['results']
+
+        with open(ROOT_DIR + '/src/preseed.json', 'r+') as f:
+            preseed_jobs = json.loads(f.read())
+        # If a job in recent runnablejobs and can't been found in the old runnablejobs,
+        # that job been determined as a new job and pick it up.
+        new_jobs = [new_job for new_job in new_data['results']
+                    if new_job not in old_data]
+        if len(new_jobs) > 0:
+            for job in new_jobs:
+                # This part of code is actually duplicate with jobtypes.
+                if job['build_system_type'] == 'buildbot':
+                    testtype = job['ref_data_name'].split(' ')[-1]
+                else:
+                    if job['ref_data_name'].startswith('desktop-test') or \
+                            job['ref_data_name'].startswith('android-test'):
+                        separator = job['platform_option'] if \
+                            job['platform_option'] != 'asan' else 'opt'
+                        testtype = job['job_type_name'].split(
+                            '{buildtype}-'.format(buildtype=separator))[-1]
+
+                    else:
+                        continue
+                for j in preseed_jobs:
+                    # If the job had already in there, we just keep it.
+                    if (j['name'] == testtype and
+                            j['platform'] == job["build_platform"] and
+                            j['buildtype'] == job["platform_option"]):
+                        continue
+
+                    else:
+                        expires = (datetime.date.today() +
+                                   datetime.timedelta(days=14)).strftime('%Y-%m-%d')
+
+                        temp = {"platform": job["build_platform"],
+                                "buildtype": job["platform_option"],
+                                "name": testtype,
+                                "expires": expires,
+                                "action": "run"}
+                        if temp not in preseed_jobs:
+                            preseed_jobs.append(temp)
```

Comment:
I really like the idea of adding new jobs into preseed for 14 days.  Possibly we should trim out expired jobs from preseed.json as well (as part of a followup patch?)